client/lib/support/support-user/pre-boot.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * External dependencies
+ */
+import localforage from 'localforage';
+import localForageBypass from './localforage-bypass';
+import debugModule from 'debug';
+
+// This module is run at page startup when support user is active. It performs
+// any pre-Calypso-boot initialization that is required by support user. Once
+// the pre-boot completes, it loads the main Calypso bundles passed in via
+// `window.supportUser.readyScripts`.
+
+const debug = debugModule( 'calypso:support-user' );
+
+const loadScript = ( script, callback ) => {
+	var bundle = document.createElement( 'script' );
+	bundle.onload = callback;
+	bundle.src = script;
+	document.body.appendChild( bundle )
+}
+
+const bootCalypso = () => {
+	if ( ! window.supportUser || !window.supportUser.readyScripts ) {
+		console.error( 'supportUserReady scripts not specified' );
+	}
+
+	const readyScripts = window.supportUser.readyScripts;
+
+	let loadedScripts = 0;
+	readyScripts.forEach( ( script ) => {
+		debug( 'Booting Calypso bundle', window.supportUserReady );
+
+		loadScript( script, () => {
+			loadedScripts++;
```

Comment:
Seems like something [`Promise.all`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all) is particularly well-suited for: 

```js
const loadScript = ( script ) => {
	return new Promise( ( resolve ) => {
		// ...
		bundle.onload = resolve;
	} );
};

Promise.all( readyScripts.map( ( script ) => loadScript( script ) ).then( window.AppBoot );
```